src/messaging/ReliableMessageMgr.cpp

```diff
@@ -350,34 +350,33 @@ bool ReliableMessageMgr::CheckAndRemRetransTable(ReliableMessageContext * rc, ui
 
 CHIP_ERROR ReliableMessageMgr::SendFromRetransTable(RetransTableEntry * entry)
 {
-    CHIP_ERROR err              = CHIP_NO_ERROR;
     ReliableMessageContext * rc = entry->rc;
-    uint32_t msgId              = 0; // Not actually used unless we reach the
-                                     // line that initializes it properly.
-
-    VerifyOrReturnError(rc != nullptr, err);
-
-    // Now that we know this is a valid entry, grab the message id from the
-    // retained buffer.  We need to do that now, because we're about to hand it
-    // over to someone else, and on failure it will no longer be available.
-    msgId = entry->retainedBuf.GetMsgId();
+    if (rc == nullptr)
+    {
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
 
     const ExchangeMessageDispatch * dispatcher = rc->GetExchangeContext()->GetMessageDispatch();
-    VerifyOrExit(dispatcher != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
-
-    err = dispatcher->SendPreparedMessage(rc->GetExchangeContext()->GetSecureSession(), entry->retainedBuf);
-    SuccessOrExit(err);
+    if (dispatcher == nullptr)
+    {
+        ChipLogError(ExchangeManager, "Crit-err %s. when sending CHIP MsgId:%08" PRIX32 ", send tries: %d",
+                     ErrorStr(CHIP_ERROR_INCORRECT_STATE), entry->retainedBuf.GetMsgId(), entry->sendCount);
```

Comment:
The error message here could be a lot more useful than just "incorrect state", now that we know _exactly_ why we are erroring out.

Reply:
same as above, potential to reuse the error message.